### ISSUE TITLE
Consumer reconnection not happening after Kafka exception

### ISSUE THREAD

**Issue by gaw** [dotnetcore/CAP#1084]
In certain situations, the consumer may be interrupted with an exception and the reconnection will not work. In my case, this happens when stopping and starting Kafka service. Service logs when the exception occurs:

```
[27-01-2022 08:37:30.278 Warning ] Transport connection is unhealthy, reconnection...
%4|1643272650.279|CONFWARN|rdkafka#producer-31| [thrd:app]: Configuration property auto.offset.reset is a consumer property and will be ignored by this producer instance
%3|1643272654.294|FAIL|rdkafka#producer-31| [thrd:kafka:9092/bootstrap]: kafka:9092/bootstrap: Failed to resolve 'kafka:9092': Name does not resolve (after 4015ms in state CONNECT)
%3|1643272654.295|ERROR|rdkafka#producer-31| [thrd:kafka:9092/bootstrap]: 1/1 brokers are down
%3|1643272654.295|ERROR|rdkafka#producer-31| [thrd:app]: rdkafka#producer-31: kafka:9092/bootstrap: Failed to resolve 'kafka:9092': Name does not resolve (after 4015ms in state CONNECT)
%3|1643272660.300|FAIL|rdkafka#producer-31| [thrd:kafka:9092/bootstrap]: kafka:9092/bootstrap: Failed to resolve 'kafka:9092': Name does not resolve (after 4020ms in state CONNECT, 1 identical error(s) suppressed)
%3|1643272660.300|ERROR|rdkafka#producer-31| [thrd:app]: rdkafka#producer-31: kafka:9092/bootstrap: Failed to resolve 'kafka:9092': Name does not resolve (after 4020ms in state CONNECT, 1 identical error(s) suppressed)
%3|1643272686.281|FAIL|rdkafka#producer-31| [thrd:kafka:9092/bootstrap]: kafka:9092/bootstrap: Connect to ipv4#172.21.0.29:9092 failed: Connection refused (after 1ms in state CONNECT)
%3|1643272686.281|ERROR|rdkafka#producer-31| [thrd:app]: rdkafka#producer-31: kafka:9092/bootstrap: Connect to ipv4#172.21.0.29:9092 failed: Connection refused (after 1ms in state CONNECT)
%3|1643272691.281|FAIL|rdkafka#producer-31| [thrd:kafka:9092/bootstrap]: kafka:9092/bootstrap: Connect to ipv4#172.21.0.29:9092 failed: Connection refused (after 0ms in state CONNECT, 5 identical error(s) suppressed)
%3|1643272691.281|ERROR|rdkafka#producer-31| [thrd:app]: rdkafka#producer-31: kafka:9092/bootstrap: Connect to ipv4#172.21.0.29:9092 failed: Connection refused (after 0ms in state CONNECT, 5 identical error(s) suppressed)
[27-01-2022 08:38:18.313 Error ] Broker: Coordinator load in progress
Confluent.Kafka.ConsumeException: Broker: Coordinator load in progress
   at Confluent.Kafka.Consumer`2.Consume(Int32 millisecondsTimeout)
   at Confluent.Kafka.Consumer`2.Consume(CancellationToken cancellationToken)
   at DotNetCore.CAP.Kafka.KafkaConsumerClient.Listening(TimeSpan timeout, CancellationToken cancellationToken)
   at DotNetCore.CAP.Internal.ConsumerRegister.<>c__DisplayClass18_1.<Execute>b__2()
```

After this error, receiving messages stops until the service that uses CAP is restarted.
Unfortunately, I can not specify the exact steps to reproduce the bug. "Coordinator load in progress" error only occurs on a machine with a large number of docker containers running and may be due to performance issues.

Judging by the source code of the ConsumerRegister class, the reconnection does not occur because field _isHealthy  is not set to false in the exception handler. What was the reason behind this code? Perhaps assigning false to field _isHealthy should be in both handlers?

```
catch (BrokerConnectionException e)
{
    _isHealthy = false;
    _logger.LogError(e, e.Message);
}
catch (Exception e)
{
    _logger.LogError(e, e.Message);
}
```

**Comment by gaw** [comment]
I'm sorry, for some reason the post was duplicated. Remove one of them please.

**Comment by yang-xiaodong** [comment]
Hello @gaw ,

Can you test #1085 is working?

**Comment by gaw** [comment]
Thanks for the quick response. I will check the fix as soon as possible.
I looked at the code changes. Perhaps it is better to additionally write errors to the log, and not just skip them with continue. Otherwise, it may make it difficult to diagnose problems.

**Comment by yang-xiaodong** [comment]
Log has been added

**Comment by gaw** [comment]
@yang-xiaodong I checked the fix. Everything works perfectly. The exception is logged and after that the connection with kafka is established.

**Comment by yang-xiaodong** [comment]
Fixed in v6.0.1
